cdci_data_analysis/analysis/instrument.py

```diff
@@ -139,41 +158,53 @@ def set_pars_from_dic(self, par_dic, verbose=False):
         else:
             param_list = [par for _query in self._queries_list for par in _query.parameters]
 
-        updated_par_dic = par_dic.copy()
+        updated_arg_dic = arg_dic.copy()
         
         for par in param_list:
             self.logger.info("before normalizing, set_pars_from_dic>> par: %s par.name: %s par.value: %s par_dic[par.name]: %s",
-                             par, par.name, par.value, par_dic.get(par.name, None))
+                             par, par.name, par.value, arg_dic.get(par.name, None))
 
             # this is required because in some cases a parameter is set without a name (eg UserCatalog),
             # or they don't have to set (eg scw_list)
             if par.name is not None and par.name not in params_not_to_be_included:
                 # set the value for par to a default format,
                 # or to a default value if this is not included within the request
-                updated_par_dic[par.name] = par.set_value_from_form(par_dic, verbose=verbose)
+                updated_arg_dic[par.name] = par.set_value_from_form(arg_dic, verbose=verbose)
 
                 if par.units_name is not None:
                     if par.default_units is not None:
-                        updated_par_dic[par.units_name] = par.default_units
+                        updated_arg_dic[par.units_name] = par.default_units
                     else:
                         raise InternalError("Error when setting the parameter %s: "
                                             "default unit not specified" % par.name)
                 if par.par_format_name is not None:
                     if par.par_default_format is not None:
-                        updated_par_dic[par.par_format_name] = par.par_default_format
+                        updated_arg_dic[par.par_format_name] = par.par_default_format
                     else:
                         raise InternalError("Error when setting the parameter %s: "
                                             "default format not specified" % par.name)
                 else:
                     self.logger.warning("units_name for the parameter %s not specified", par.name)
 
             self.logger.info("after normalizing, set_pars_from_dic>> par: %s par.name: %s par.value: %s par_dic[par.name]: %s",
-                             par, par.name, par.value, par_dic.get(par.name, None))
+                             par, par.name, par.value, arg_dic.get(par.name, None))
 
             if par.name == "scw_list":
                 self.logger.info("set_pars_from_dic>> scw_list is %s", par.value)
 
-        return updated_par_dic
+        self.allow_unknown_arguments = self.allow_unknown_arguments or arg_dic.get('allow_unknown_args', 'False') == 'True'
```

Comment:
it seems intentional that if the attribute is already set to True, the argument is ignored. But why?

Reply:
true, I hadn't seen it in that way...perhaps it is better to set it with the value within the `arg_dic` if present, otherwise it doesn' change...but it depends on which one has the "priority"...